Aha, I dumped a big AST to JSON, searched for a `CXXBindTemporaryExpr` inside a `VarDecl` and came up with this example:

```
/Users/doug/Desktop/makestring.cpp:10:7: warning: function with 'nonblocking' attribute must not call non-'nonblocking' destructor 'String::~String' [-Wfunction-effects]
   10 |         auto str = String::make();
      |              ^
/Users/doug/Desktop/makestring.cpp:10:13: warning: function with 'nonblocking' attribute must not call non-'nonblocking' destructor 'String::~String' [-Wfunction-effects]
   10 |         auto str = String::make();
      |                    ^
```

It may be a little annoying to get two warnings but technically I think there really are two implicit calls to the destructor. (right? the result of `String::make()` might be getting moved-from but there's still an implicit destructor.)

--- a/clang/docs/ReleaseNotes.rst
+++ b/clang/docs/ReleaseNotes.rst
@@ -452,6 +452,7 @@ Bug Fixes to Attribute Support
 - Fix a crash when the function name is empty in the `swift_name` attribute. (#GH157075)
 - Fixes crashes or missing diagnostics with the `device_kernel` attribute. (#GH161905)
 - Fix handling of parameter indexes when an attribute is applied to a C++23 explicit object member function.
+- Fixed several false positives and false negatives in function effect (`nonblocking`) analysis. (#GH166078) (#GH166101) (#GH166110)
 
 Bug Fixes to C++ Support
 ^^^^^^^^^^^^^^^^^^^^^^^^

